### PR TITLE
inserted maxlength attribute for title field

### LINKED ISSUE #5426: Limit extra long wiki titles
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

There is a small section of wiki title creation that hasn't been taken care of. A wiki title needs to be as brief as any other title. However there are no limits to this when creating a wiki. http://localhost:3000/wiki  Our title can, in worst case, look like this:

long-wiki

We can add an HTML attribute `maxlength=""` to the title field and set it equal to the number of characters you are limiting to. In our case we can limit title characters to 50 or 60. Your choice.

The target file and line number is: 
https://github.com/publiclab/plots2/blob/master/app/views/wiki/edit.html.erb#L61

Add the attribute to this input tag and test if it is working. Note that only admins can create new Wikis so you will have to log in as an admin if you want to see this play on your local server. All the best! :smile: 

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by kvenkman**
Fixes #5426 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by kvenkman** [comment]
Could folks from @publiclab verify that the wiki titles are now limited to 50 characters?

**Comment by IshaGupta18** [comment]
Could you please attach a screenshot of the feature?

**Comment by kvenkman** [comment]
Sorry, I can't get the code up and running on my local machine because of a Ruby versioning error. I'll uncheck that box in my original comment.

File: app/views/wiki/edit.html.erb [hunk 1]
```diff
@@ -57,7 +57,7 @@
           <input class="form-control" id="taginput" type="hidden" name="tags" value="<%= params[:tags] %>" />
 
           <div class="form-group">
-            <input tabindex="1" name="title" type="text" class="form-control" id="title" value="<% if @node && @node.latest %><%= @node.latest.title %><% else %><%= params[:title] %><% end %>">
+            <input tabindex="1" name="title" type="text" class="form-control" id="title" maxlength="50" value="<% if @node && @node.latest %><%= @node.latest.title %><% else %><%= params[:title] %><% end %>">
```
Comment (cesswairimu) [inline app/views/wiki/edit.html.erb]: Hey @kvenkman, while this is great and its what was described on the issue. I would suggest we also add the validation on the active record level.
 Please change this line https://github.com/publiclab/plots2/blob/master/app/models/node.rb#L97 to 
```
  validates :title, presence: true, length: { minimum: 3, maximum: 50 }
```

Comment (kvenkman) [reply app/views/wiki/edit.html.erb]: I've commited changes to my branch - does that work?

Comment (cesswairimu) [reply app/views/wiki/edit.html.erb]: Yep it does..Thanks

**Comment by cesswairimu** [comment]
Hey @jywarren, we are adding a maximum length for  a title so that we don't have so long titles..Is 50 chars a reasonable length or which length would be great for the nodes? Thanks

**Comment by jywarren** [comment]
I think this may be a question for @publiclab/community-reps -- as to whether we want to limit it!

**Comment by SidharthBansal** [comment]
@jywarren how shall we proceed here?

**Comment by jywarren** [comment]
I'm afraid I think we should drop this, as it's not clear we have community
consensus around limiting title length. Thank you!

**Comment by SidharthBansal** [comment]
Feel Free to reopen it if needed
Thanks all

**Pull request closed without merging**